e2e/tests/test_e2e.py

```diff
@@ -227,30 +270,38 @@ def test_infrastructure_roles(self):
         }
         k8s.update_config(patch_infrastructure_roles)
 
-        # wait a little before proceeding
-        time.sleep(30)
-
         try:
             # check that new roles are represented in the config by requesting the
             # operator configuration via API
-            operator_pod = k8s.get_operator_pod()
-            get_config_cmd = "wget --quiet -O - localhost:8080/config"
-            result = k8s.exec_with_kubectl(operator_pod.metadata.name, get_config_cmd)
-            roles_dict = (json.loads(result.stdout)
-                          .get("controller", {})
-                          .get("InfrastructureRoles"))
-
-            self.assertTrue("robot_zmon_acid_monitoring_new" in roles_dict)
-            role = roles_dict["robot_zmon_acid_monitoring_new"]
-            role.pop("Password", None)
-            self.assertDictEqual(role, {
-                "Name": "robot_zmon_acid_monitoring_new",
-                "Flags": None,
-                "MemberOf": ["robot_zmon"],
-                "Parameters": None,
-                "AdminRole": "",
-                "Origin": 2,
-            })
+
+            def verify_role():
+                operator_pod = k8s.get_operator_pod()
+                get_config_cmd = "wget --quiet -O - localhost:8080/config"
+                result = k8s.exec_with_kubectl(operator_pod.metadata.name, get_config_cmd)
+                try:
+                    roles_dict = (json.loads(result.stdout)
+                                .get("controller", {})
+                                .get("InfrastructureRoles"))
+                except:
+                    return False
+
+                if "robot_zmon_acid_monitoring_new" in roles_dict:
+                    role = roles_dict["robot_zmon_acid_monitoring_new"]
+                    role.pop("Password", None)
+                    self.assertDictEqual(role, {
+                        "Name": "robot_zmon_acid_monitoring_new",
+                        "Flags": None,
+                        "MemberOf": ["robot_zmon"],
+                        "Parameters": None,
+                        "AdminRole": "",
+                        "Origin": 2,
+                    })
+                    return True
+                else:
```

Comment:
the `else` is not needed; simply `return False` will do

Reply:
fixed locally